app/models/ability.rb

```diff
@@ -35,7 +35,42 @@ def initialize(user)
     #   end
 
     user ||= User.new # guest user (not logged in)
+    id = user.id
+
     # All users can only update their own user attributes
-    can :update, User, id: user.id
+    alias_action :create, :read, :update, :destroy, :to => :crud
+
+    can :crud, Team
+    can :update, User, id: id
+
+    can :assign_ownership, Team, Team do |team|
+      team.owners.include? user
+    end
+
+    can :delete_ownership, Team, Team do |team|
+      team.owners.include? user
+      team.owners.length > 1
+    end
+
+    can :delete_membership, Team, Team do |team|
+      team.owners.include? user and Ability.num_owners(team, team_member) > 0
+    end
+
+    can :delete_membership, Team, Team do |team|
+      Ability.num_owners(team, team_member) > 0 and Integer(id) == Integer(team_member)
```

Comment:
Regarding `Integer(id)`: It seems to me that `id` is already an integer, isn't it? This would avoid at least one of the two conversions. The other conversion seems to be necessary because the POST parameter `team_member` is a string – right?

Reply:
yup -- just making sure here; but you're right; technically `id`'s fine as it is.